src/components/NavbarComponent.js

```diff
@@ -0,0 +1,106 @@
+import { POPUP_DISAPPEAR_TIMEOUT } from "../Constants";
+import {
+  EventHandlerService,
+  PDFLEvents,
+} from "../services/EventHandlerService";
+
+/**
+ * Component that takes in the PDF file that user uploads and processes it.
+ *
+ * @property {Object} components object that holds DOM elements that are within component
+ * @property {HTMLElement} components.openNew button to click for upload a new pdf from the reader component
+ * @property {HTMLElement} components.loader loader for showing the pdf uploaded
+ * @property {HTMLElement} components.errorMessage error message for a wrong pdf uploaded from the pdf reader component
+ */
+class NavbarComponent {
+  components = {
+    openNew: document.querySelector("#open-new-pdf"),
+    loader: document.querySelector("#loader"),
+    errorMessage: document.querySelector(
+      "#message-wrong-type-fileupload-reader"
+    ),
+  };
+
+  /**
+   * Creates and initializes navbar component.
+   * @constructor
+   */
+  constructor() {
+    this.#registerEvents();
+  }
+
+  /**
+   * Adds event listeners to ioeb a new pdf button from the pdf reader component and the shortcuts
+   * @private
+   */
+  #registerEvents = () => {
+    this.components.openNew.addEventListener("input", this.#onNewFile);
+
+    EventHandlerService.subscribe(
+      PDFLEvents.onKeyboardKeyDown,
+      (functionalKeys, key) => {
```

Comment:
We're trying to write event handlers in registerEvents method such as:
this.some_element.addEventListener("event", this.#some_function);
because in that way register events **only registers events** and #some_function only does some logic on event which is then described in function documentation.
Now youre not aware that this class does something on onKeyboardKeyDown event because there is no function in this class that is triggered on that event, as well as that now registerEvents is now not only registering events -> one function should always only do one thing.